### PR TITLE
Team6 Food Intake Strategy Update

### PR DESCRIPTION
**Pull request by MattScottEEE**
# Summary

Updated food intake strategies based on new HP functions in main

## Additional Information

Agents take food according to social motive and HP function

## Test Plan

No conflicts with infra files - logs have been run to confirm that behaviours are correct

### CODE REVIEW

File: pkg/agents/team6/strategies.go [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
 
+type levelsData struct {
+	strongLevel  float64
+	healthyLevel float64
+	weakLevel    float64
+	critLevel    float64
+}
+
+var daysInCritical int = 0
+var foodTakeDay int
```
Comment (jzzheng22) [inline pkg/agents/team6/strategies.go:22]:
```suggestion
daysInCritical := 0
foodTakeDay := 0
```

File: pkg/agents/team6/strategies.go [hunk 2]
```diff
@@ -1,24 +1,93 @@
...
+
 func (a *CustomAgent6) foodIntake() float64 {
-	thresholds := thresholdData{satisficeThresh: 20.0, satisfyThresh: 60.0, maxIntake: 80.0}
+
```
Comment (jzzheng22) [inline pkg/agents/team6/strategies.go:25]:
```suggestion
```

File: pkg/agents/team6/strategies.go [hunk 3]
```diff
@@ -1,24 +1,93 @@
...
+	thresholds := thresholdData{
+		maxIntake: 80.0,
+	}
+
+	levels := levelsData{
+		strongLevel:  0.6 * float64(healthInfo.MaxHP),
+		healthyLevel: 0.3 * float64(healthInfo.MaxHP),
+		weakLevel:    0.1 * float64(healthInfo.MaxHP),
+		critLevel:    0.0,
+	}
```
Comment (jzzheng22) [inline pkg/agents/team6/strategies.go:37]: Should this be an int?

File: pkg/agents/team6/strategies.go [hunk 4]
```diff
@@ -1,24 +1,93 @@
...
+	case "Narcissist": // Eat max intake (Possible TODO: Stay in strong instead?)
 		return thresholds.maxIntake
+
 	default:
 		return 0.0
 	}
 }
+
+func foodRequired(currentHP, goalHP float64, healthInfo *health.HealthInfo) float64 {
+	return healthInfo.Tau*math.Log(healthInfo.Width) - healthInfo.Tau*math.Log(healthInfo.Width-goalHP+0.75*currentHP-10+0.25*float64(healthInfo.WeakLevel))
```
Comment (jzzheng22) [inline pkg/agents/team6/strategies.go:80]:
```suggestion
        denominator := healthInfo.Width-goalHP+0.75*currentHP-10+0.25*float64(healthInfo.WeakLevel
	return healthInfo.Tau*math.Log(healthInfo.Width / denominator)
```

Comment (jzzheng22) [reply pkg/agents/team6/strategies.go:80]: Potential simplification? (Check my maths though!)

**Review by jzzheng22 — CHANGES_REQUESTED** [review]

File: pkg/agents/team6/strategies.go (hunk 1 above)
Comment (MattScottEEE) [reply pkg/agents/team6/strategies.go:22]: We need these both outside the function, as they get updated inside the function. I think you can only use short assign inside of functions?

File: pkg/agents/team6/strategies.go (hunk 4 above)
Comment (MattScottEEE) [reply pkg/agents/team6/strategies.go:79]: Ooh nice nice

File: pkg/agents/team6/strategies.go (hunk 1 above)
Comment (jzzheng22) [reply pkg/agents/team6/strategies.go:22]: nonononono no global variables

**Review by jzzheng22 — CHANGES_REQUESTED** [review]
No global variables at all, unless they are consts or singletons. Absolutely no global variables.

File: pkg/agents/team6/strategies.go (hunk 3 above)
Comment (MattScottEEE) [reply pkg/agents/team6/strategies.go:34]: Yes, but would require a huge amount of casting both in struct and when accessed in functions

**Review by jzzheng22 — APPROVED** [review]

**Pull request merged**